blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/include/HttpTxManager.h#L56
```
 typedef std::map<std::string, XAWrapper*> XABranchMap;
 XABranchMap _branches;
```
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/include/XAResourceManager.h#L61
```
typedef std::map<XID, XAWrapper *> XABranchMap;
XABranchMap branches_;
```
It looks like duplicated ?